### PR TITLE
Provide explanation of dangling indices, fixes #26008

### LINKED ISSUE #26008: [DOCS] Provide an explanation of what dangling indices are and how they can occur
The elasticsearch docs make reference to dangling indices:
- https://www.elastic.co/search?q=dangling&section=Learn%2FDocs%2FElasticsearch%2FReference%2F5.5

but there's no explanation of what they are or how they come about.
It would be good to have a glossary of some type to refer to for various terminology like this that could be expanded over time.

It would be good to link/reference [Index Tombstones](https://www.elastic.co/guide/en/elasticsearch/reference/5.5/misc-cluster.html#cluster-max-tombstones) when documenting dangling indices too.

### PR DESCRIPTION
**Pull request by geekpete**
Adding explanation of dangling indices back to the docs to fix #26008

### CODE REVIEW

File: docs/reference/modules/gateway.asciidoc
Comment (jasontedor) [inline docs/reference/modules/gateway.asciidoc]: This should refer to shards only.

Comment (jasontedor) [reply docs/reference/modules/gateway.asciidoc]: I do not see a reason to format "data/" here, we can say "data directory".

**Review by jasontedor — CHANGES_REQUESTED** [review]
I left a few comments.

**Comment by geekpete** [comment]
Thanks, I've adjusted with these suggestions.

File: docs/reference/modules/gateway.asciidoc
Comment (debadair) [inline docs/reference/modules/gateway.asciidoc]: There's an extra "directory" in the first sentence. 

I recommend avoiding future tense unless it's really necessary--and in this case I think it would be clearer to stick to present tense. I'd make the following tweaks:

When a node joins the cluster, any shards stored in its local data directory 
that do not exist in the cluster are imported into the cluster. This functionality
is intended as a best effort to help you recover if a cluster loses all master
nodes. If a new master node is started that is unaware of the other indices in
the cluster, the old indices are imported as the old nodes join the cluster, not
deleted.

I reflexively use "that" instead of "which" for dependent clauses--though that "rule" isn't as hard and fast as some folks like to think.

We normally just keep pushing updates to doc PRs until it's ready to merge, then squash to a single commit and merge. Occasionally we'll keep multiple commits when merging, particularly if there are changes that are related, but not reflected in original commit messages.

LGTM as long as the directory stutter is fixed.

**Review by bleskes — APPROVED** [review]
LGTM

**Comment by geekpete** [comment]
@jasontedor I think I merged this one too quickly, can you review whenever you get some spare time, no rush on this either. Thanks

File: docs/reference/modules/gateway.asciidoc
Comment (jasontedor) [inline docs/reference/modules/gateway.asciidoc]: The word `directory` is duplicated from the previous line.

**Review by jasontedor — COMMENTED** [review]
Thanks for the ping, I missed the previous one (sorry). I see that you merged but I left a comment about a duplicated word.

**Pull request merged**
